synthesizer/src/vm/helpers/cost.rs

```diff
@@ -23,6 +23,14 @@ use synthesizer_program::{Command, Finalize, Instruction};
 
 use std::collections::HashMap;
 
+#[allow(unused)]
+/// Helper struct for i.a. SDK to return costs for users.
+pub struct ExecutionCosts {
+    storage_cost: u64,
+    finalize_cost: u64,
+    execution_cost: u64,
+}
+
```

Comment:
Looks good, just needs accessor methods for end users, and perhaps a display method for printing it.

```suggestion
impl ExecutionCosts {
    /// Cost for storing the execution transaction on the Aleo Network
    pub fn storage_cost() -> u64 {
        self.storage_cost
    }
    
     /// Cost for executing all operations within the function's finalize scope on the Aleo Network
    pub fn finalize_cost() -> u64 {
        self.finalize_cost
    }
    
     /// Cost for verification of the function execution
    pub fn execution_cost() -> u64 {
        self.storage_cost
    }
}
```

Reply:
I feel like a rebel today: cc10cce5e